src/main/java/com/incentives/piggyback/offers/serviceImpl/OfferServiceImpl.java

```diff
@@ -109,4 +103,34 @@ public String sendNotification(BroadcastRequest broadcastRequest) {
 		return response.getBody().getData();
 	}
 
+	@Override
+	public List<UserData> getUsersWithInterest(List<Long> users, List<String> interests) {
+		String url = env.getProperty("users.api.usersWithInterest");
+		HttpHeaders headers = new HttpHeaders();
+		headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
```

Comment:
Is this working? You should be generating a login request and passing Bearer token to the User API.

Reply:
I have implemented a secret key in headers for server to server interaction

Reply:
Added a new story to implement Secret Key auth for User service for API-API interactions.